pkg/storage/chunk/cache/redis_client.go

```diff
@@ -96,6 +83,49 @@ func NewRedisClient(cfg *RedisConfig) (*RedisClient, error) {
 	}, nil
 }
 
+func deriveEndpoints(endpoint string, lookup func(host string) ([]string, error)) ([]string, error) {
+	if lookup == nil {
+		return nil, fmt.Errorf("lookup function is nil")
+	}
+
+	endpoints := strings.Split(endpoint, ",")
+	if len(endpoints) < 1 {
+		return endpoints, nil
+	}
+
```

Comment:
should we also return early if `len(endpoints) > 1`
I am assuming the special case handling below is only when a single endpoint is provided?

Reply:
That's a good call